hal/src/nRF52840/adc_hal.cpp

```diff
@@ -138,7 +139,11 @@ void hal_adc_dma_init() {
         adcReference = HAL_ADC_REFERENCE_INTERNAL;
     }
 #else
-    (void)adcReference;
+    hal_device_hw_info hwInfo = {};
+    hal_get_device_hw_info(&hwInfo, nullptr);
+    if ((hwInfo.features & HW_FEATURE_FLAG_USE_INTERNAL_ADC_REFERENCE_BIT) == 0) {
+        adcReference = HAL_ADC_REFERENCE_INTERNAL;
+    }
```

Comment:
Do we want to check this feature flag on `PLATFORM_BORON` as well? In case we move forward with a HW version 4 or something?